### PR TITLE
0009562: Add new account functions

### PR DESCRIPTION
**Pull request by emre1702**
Added getAccountIP & getAccountsByIP.
Added getAccountsByData.
Added setAccountName.

string getAccountIP ( account theAccount )
table getAccountsByIP ( string theIP )
table getAccountsByData ( string dataName, string value )
boolean setAccountName ( account theAccount, string newName[, bool allowCaseVariations = false] )
 
Bugtracker: https://bugs.mtasa.com/view.php?id=9562

Also with setAccountName this code will get useless:
https://wiki.multitheftauto.com/wiki/SetAccountName

### CODE REVIEW

**Comment by emre1702** [comment]
Ok, that's it.

**Comment by CrosRoad95** [comment]
add option to save accounts in mysql :)

**Comment by emre1702** [comment]
Seems like I got some problems with pull requests and GitHub in Visual Studio.
The last three commits shouldn't be there, tried to remove them, made it worse.
If you can, please delete them, they belong to another pull request :/

**Comment by qaisjp** [comment]
Hi @emre1702,

To prevent this from happening, you need to make a branch for each pull request you'd like to create.

To fix your branch, run these commands in the command line:

```
git checkout master
git reset ef3b7167e011c6 --hard
git config --global pull.rebase true
git push --force
```

Let me know if you have any problems.

**Comment by emre1702** [comment]
Thx, worked.
Learned pull requests after this request, this was my first.
The other ones worked without problems 😆 
Got no problems anymore ^^

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 1]
```diff
@@ -11332,6 +11332,34 @@ bool CStaticFunctionDefinitions::RemoveAccount ( CAccount* pAccount )
...
+    assert ( pAccount );
+    assert ( !strNewName.empty() );
+
+    if ( pAccount->IsRegistered () ) {
+            // Check for case variations if not allowed
+        if ( !bAllowCaseVariations ) {
+            SString strCaseVariation = m_pAccountManager->GetActiveCaseVariation( strNewName );
+            if ( !strCaseVariation.empty() ) {
+                strOutError = SString( "Already an account using a case variation of that name ('%s')", *strCaseVariation );
+                return NULL;
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Should not be returning `NULL` here, should be returning `false`.

**Comment by emre1702** [comment]
@qaisjp Right, sry.
Copied from AddAccount, but here a null makes no sense.

**Comment by ccw808** [comment]
What is the reason for adding account ID access?

**Comment by Dezash** [comment]
@ccw808 it is an easy way to reference accounts on systems that cannot handle certain characters. You do not need to make any conversions this way.

**Comment by ccw808** [comment]
I am concerned about exposing sqlite generated row ids. Please split the account ID access stuff into separate pull request.

**Comment by emre1702** [comment]
@ccw808 Done

**Comment by ccw808** [comment]
Thanks.
Please change the curly brace style so it is consistent with the existing code.

**Comment by emre1702** [comment]
@ccw808 Done

**Comment by ccw808** [comment]
CStaticFunctionDefinitions::SetAccountName and CLuaAccountDefs::SetAccountName need curly brace changes

**Comment by emre1702** [comment]
Oh sry, forgot setAccoutName.
Fixed @ccw808

**Comment by ccw808** [comment]
Thanks

**Pull request merged**